actionpack/test/controller/routing_test.rb

```diff
@@ -1604,6 +1604,18 @@ def test_route_with_subdomain_and_constraints_must_receive_params
     assert_equal(name_param, 'mypage')
   end
 
+  def test_if_routes_in_scope_with_optional_path_processed_after_root
+    set.draw do
+      scope path: '(:name)', constraints: lambda { |request| request.params[:name].blank? || (request.params[:name] == 'foo') } do
+        root 'pages#index'
+        get 'bar' => 'pages#bar'
+      end
+    end
+
+    assert_equal({:controller => 'pages', :action => 'bar'},
```

Comment:
New tests should use new hash syntax: `assert_equal({controller: 'pages', action: 'bar'}`